sdk/samples/iot/hub/esp8266nodemcu/azure_iot_hub_telemetry.ino

```diff
@@ -92,7 +92,7 @@ void receivedCallback(char* topic, byte* payload, unsigned int length)
   Serial.print("Received [");
   Serial.print(topic);
   Serial.print("]: ");
-  for (int i = 0; i < length; i++)
+  for (int i = 0; i <= length; i++)
```

Comment:
This is not correct, it would lead to reading beyond the payload size, resulting in a memory access failure.